Framework/Geometry/src/Instrument/CompAssembly.cpp

```diff
@@ -305,6 +306,9 @@ CompAssembly::getComponentByName(const std::string &cname, int nlevels) const {
     }
 
     auto rectDet = boost::dynamic_pointer_cast<const RectangularDetector>(node);
+    auto structDet =
+        boost::dynamic_pointer_cast<const StructuredDetector>(node);
```

Comment:
This seems wrong to me that the parent is aware of the derived types. I do however appreciate that this was happening already. Would be nice to clean this up in the future.